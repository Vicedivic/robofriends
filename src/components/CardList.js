import React from "react";
import Card from "./Card";

const CardList = ({ robots }) => {
  return (
    <div>
      {robots.map(robot => (
        <Card {...robot} key={robot.id} />
      ))}
    </div>
  );
};

export default CardList;
