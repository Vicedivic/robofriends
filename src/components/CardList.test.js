import React from "react";
import ShallowRenderer from "react-shallow-renderer";
import { shallow } from "enzyme";
import CardList from "./CardList";
import robots from "../robots";

describe("CardList", () => {
  const renderer = new ShallowRenderer();
  const cardlistHtml = renderer.render(<CardList robots={robots} />);
  const cardlist = shallow(<CardList robots={robots} />);

  it("renders properly", () => {
    expect(cardlistHtml).toMatchSnapshot();
  });

  it("renders Cards", () => {
    expect(cardlist.find("Card").exists()).toBe(true);
  });
});