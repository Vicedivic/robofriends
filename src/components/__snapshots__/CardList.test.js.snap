// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList renders properly 1`] = `
<div>
  <Card
    email="sincere@april.biz"
    id={1}
    name="Leanne Graham"
  />
  <Card
    email="shanna@melissa.tv"
    id={2}
    name="Ervin Howell"
  />
  <Card
    email="nathan@yesenia.net"
    id={3}
    name="Clementine Bauch"
  />
  <Card
    email="julianna.oconner@kory.org"
    id={4}
    name="Patricia Lebsack"
  />
</div>
`;
